server/app/services/program/ProgramService.java

```diff
@@ -781,10 +781,34 @@ public ErrorAnd<ProgramDefinition, CiviFormError> updateLocalization(
     ProgramDefinition programDefinition = getFullProgramDefinition(programId);
     ImmutableSet.Builder<CiviFormError> errorsBuilder = ImmutableSet.builder();
     validateProgramText(errorsBuilder, "display name", localizationUpdate.localizedDisplayName());
-    validateProgramText(
-        errorsBuilder, "short display description", localizationUpdate.localizedShortDescription());
+
+    // only validate the short description if the program has one
+    // short description is now a required field, but older programs might not have one yet
+    Optional<String> shortDescription =
+        programDefinition.localizedShortDescription().maybeGet(LocalizedStrings.DEFAULT_LOCALE);
+    if (shortDescription.isPresent() && !shortDescription.get().isBlank()) {
+      validateProgramText(
+          errorsBuilder,
+          "short display description",
+          localizationUpdate.localizedShortDescription());
+    }
+
     validateBlockLocalizations(errorsBuilder, localizationUpdate, programDefinition);
 
+    // only validate application steps if the program has them and is not a common intake program
+    // one application step is now required, but older programs might not have one yet
+    if (!programDefinition.isCommonIntakeForm()
+        && !programDefinition.applicationSteps().isEmpty()) {
```

Comment:
I couldn't think of a way to test this in a unit test (that validation is skipped if there are no application steps), but I tested it manually by setting the value for `application_steps` in my db to the default `[]`. Same with short description above.